fix pl_worker_init_fn for pytorch 1.9

## 🐛 Bug

Investigate a test.

### To Reproduce

Run this test:

```
FAILED tests/trainer/test_dataloaders.py::test_missing_worker_init_fn - asser...
= 1 failed, 1422 passed, 263 skipped, 3 xfailed, 2760 warnings in 435.18s (0:07:15) =
```

### Expected behavior

obvious. 

### Environment

PyTorch 1.9 (nightly)

--- a/tests/trainer/test_dataloaders.py
+++ b/tests/trainer/test_dataloaders.py
@@ -803,8 +803,12 @@ def _user_worker_init_fn(_):
     pass
 
 
+@RunIf(max_torch="1.8.9")
 def test_missing_worker_init_fn():
-    """ Test that naive worker seed initialization leads to undesired random state in subprocesses. """
+    """
+    Test that naive worker seed initialization leads to undesired random state in subprocesses.
+    PyTorch 1.9+ does not have this issue.
+    """
     dataset = NumpyRandomDataset()
 
     seed_everything(0)

